chain-evm/src/transaction.rs

```diff
@@ -0,0 +1,49 @@
+use chain_core::{
+    packer::Codec,
+    property::{Deserialize, ReadError, Serialize, WriteError},
+};
+use ethereum::TransactionV2;
+use rlp::{decode, Decodable, DecoderError, Encodable, Rlp, RlpStream};
+use typed_bytes::ByteBuilder;
+
+/// Wrapper type for `ethereum::TransactionV2`, which includes the `EIP1559Transaction`, `EIP2930Transaction`, `LegacyTransaction` variants.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct EthereumTransaction(TransactionV2);
```

Comment:
I assume we should also add implementation of the `TryFrom`/`TryInto` for the conversion into the `EvmTransaction` type